homeassistant/components/bsblan/climate.py

```diff
@@ -0,0 +1,230 @@
+        if not self._info.device_identification:
+            return self._info.controller_family
+        return self._info.device_identification
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return self._info.device_identification
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement which this thermostat uses."""
+        if self._temperature_unit == "&deg;C":
+            return TEMP_CELSIUS
+        return TEMP_FAHRENHEIT
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return the current operation mode."""
+        return self._current_hvac_mode
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def preset_modes(self):
+        """List of available preset modes."""
+        return self._preset_modes
+
+    @property
+    def preset_mode(self):
+        """Return the preset_mode."""
+        if self._current_hvac_mode == 2:
+            return PRESET_ECO
+        return PRESET_NONE
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set HVAC mode."""
+        _LOGGER.info("Setting HVAC mode to: %s", hvac_mode)
+        await self.async_set_data(hvac_mode=hvac_mode)
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        await self.async_set_data(**kwargs)
+
+    async def async_set_data(self, **kwargs: Any) -> None:
+        """Set device settings using BSBLan."""
+        data = {}
+
+        if ATTR_TEMPERATURE in kwargs:
+            data[ATTR_TARGET_TEMPERATURE] = kwargs[ATTR_TEMPERATURE]
+            _LOGGER.info("data = %s", data)
+
+        if ATTR_HVAC_MODE in kwargs:
+            data[ATTR_HVAC_MODE] = HA_STATE_TO_BSBLAN[kwargs[ATTR_HVAC_MODE]]
+            _LOGGER.info("data hvac mode = %s", data)
+
+        try:
+            await self.bsblan.thermostat(**data)
+        except BSBLanError:
+            _LOGGER.error("An error occurred while updating the BSBLan Device")
+            self._available = False
+
+    async def async_update(self) -> None:
+        """Update BSBlan entity."""
+        try:
+            state: State = await self.bsblan.state()
+        except BSBLanError:
+            if self._available:
+                _LOGGER.error("An error occurred while updating the BSBLan device.")
```

Comment:
We don't end logging messages with period.
```suggestion
                _LOGGER.error("An error occurred while updating the BSBLan device")
```